### PR TITLE
Don't show tooltips when target is not visible in sidebar scroll container

### PR DESCRIPTION
**Pull request by jasontk19**
## Description

Fixes # 2469.

* Bring in `react-visibility-sensor`
* ~~Prevent webpack from transpiling `react-visibility-sensor` [more here](https://github.com/joshwnj/react-visibility-sensor/issues/148)~~
* Apply to palette legend tooltips so they don't show when the tooltip target is not within the visible area of the sidebar

### CODE REVIEW

File: web/js/components/sidebar/paletteLegend.js [hunk 1]
```diff
@@ -31,6 +33,11 @@ class PaletteLegend extends React.Component {
 
   componentDidMount() {
     this.updateCanvas();
+    this.setState(() => {
+      return {
+        scrollContainerEl: document.querySelector('#productsHolder .simplebar-wrapper')
```
Comment (Benjaki2) [inline web/js/components/sidebar/paletteLegend.js]: I think this needs to be relative to the active compare side to get it to work when the B state is active.

**Comment by jasontk19** [comment]
@edplato maybe a dumb question but... did you stop the `npm run watch` script and restart it?  I see the above error when switching to this branch while `watch` is running but stopping it, then running `npm ci` then `npm run watch` seems to work for me.

**Comment by edplato** [comment]
Changing the import to `import VisibilitySensor from 'react-visibility-sensor/visibility-sensor';` allows it to build and run.

**Comment by edplato** [comment]
@jasontk19 Yeah, I exited `watch` mode, removed `node_modules` (after first fail just in case), ran `npm install`, and `npm run build` / `IGNORE_ERRORS=true npm run build`.

**Comment by jasontk19** [comment]
Ok looks like it's a Windows vs Mac thing... the Travis builds for Windows are failing for that exact reason.  I'll take a look.

File: web/js/components/sidebar/paletteLegend.js (hunk 1 above)
Comment (jasontk19) [reply web/js/components/sidebar/paletteLegend.js]: Fixed 👍

**Comment by jasontk19** [comment]
There are a number of e2e tests failing due to the change that causes the layer list to not be rendered if it's not active.  I'm going through them now to update the tests and confirm nothing is actually broken by this change.

File: web/js/components/sidebar/paletteLegend.js (hunk 1 above)
Comment (edplato) [reply web/js/components/sidebar/paletteLegend.js]: I needed to change the selector to just `#productsHolder` to make it work - with `.simplebar-wrapper` included I would still see the tooltip on hover even with sidebar collapsed.

Comment (jasontk19) [reply web/js/components/sidebar/paletteLegend.js]: It sounds like you might be seeing the issue that was fixed in #2480?

**Pull request merged**